penpot/penpot
# Resize and Transforms using WebAssembly

How to verify it:

Follow these steps with a version of Penpot with and without WebAssembly (try to spend almost the same time doing the same things):
- 1. Enter workspace (open any file, with huge files is easier to see metrics).
- 2. Open DevTools > Options (⋮) > More Tools > JavaScript Profiler.
- 3. Go to JavaScript Profiler > Record (:red_circle:).
- 4. Apply some transformations to complex boards.
- 5. Stop recording.

After that verify that the following times (%) have decreased:
- (garbage collector)
- `app$common$types$modifiers$modifiers__GT_transform` (`app.common.types.modifiers.cljs`: `modifiers->transform`)
- `app$main$data$workspace$transforms$start_resize_$_resize` (`app.main.data.workspace.transforms.cljs`: `start-resize`)

Extra features:
- Now in the console you can write: `debug.dump_resize_wasm()` and `debug.dump_transform_wasm()` to get info about the loaded WebAssembly modules.
